### DISCUSSION TITLE
How to run longhorn as non-root/rootless?

### DISCUSSION THREAD

**Discussion by ncecsc** [longhorn/longhorn#6118] [Q&A]
Hey all,

I'm working on a bare-metal Kubernetes project that requires CIS & PCI hardening for all aspects of the project, and one of them is to not have any rooted pods/containers.

I'm planning on using Longhorn as the data backend but I'm struggling to find documentation for running Longhorn as non-root, is this possible to do? What would change in the installation? If not, what would you recommend as an alternative to Longhorn?

I'm looking to use Usernetes or K3s as the base for running Kubernetes components as non-root also, if that changes anything.

Thanks in advance!

**Comment by Akay7** [comment]
[Longhorn required](https://longhorn.io/docs/1.12.1/deploy/install/#root-and-privileged-permission) permissions and privileges by design. And CIS doesn't required to remove all priviliges, it just asks to not give priviliges that aren't in use.

 _But if you not use RWX, volume encryption, and few other things. Then I think it's possible to minimize permissions list._
